test/com/google/enterprise/adaptor/filenet/PermissionsTest.java

```diff
@@ -691,4 +694,275 @@ public void testMarking_UserNotMatchingAnyAces() {
         SecurityPrincipalType.USER, "user2@bar.example.com", user1, false,
         AccessRight.NONE);
   }
+
+  //======================================================================
+
+  @SuppressWarnings("deprecation")  // For PermissionSource.MARKING
+  @Test
+  public void testMarkingAcl_emptyAcl_allowConstraintMask() throws Exception {
+    Permissions testPerms = new Permissions(perms);
```

Comment:
Maybe s/perms/new AccessPermissionListMock()/ when we just need it as an empty list.